### PR TITLE
Open search bar on find next/find previous to avoid permanent code highlights

### LINKED ISSUE #20643: Pressing F3 in text editor when search field is closed makes permanent search highlights that you need CTRL + F to disable
Sooo... this issue requires a set of actions to appear, but paradoxically, it happened to me enough times to get me annoyed and I finally found a way to reproduce it.
tl;dr it causes a permanent search highlight to appear and you need to CTRL + F to cancel it.

**Steps to reproduce**

0. Here's example code to test it:
```
func _ready():
	var a = 1
	var b = 1
	var c = 1
```
1. Press CTRL + F
2. Type "1". This will highlight all 1s in your editor
image
3. Press ESC to close the search text field. It will correctly remove highlights
image
4. Press F3. The highlights are back, but there's no search field
image
You can't remove the highlights now. Pressing ESC does nothing, because there's no search field to close. You need to press CTRL + F to re-enable search field and then press ESC.

My suggested fixes:
-make F3 bring back search field if it's closed
OR
-make ESC always cancel search highlights even if search field is invisible

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #20643

I went with the first fix proposed there, which is forcing search bar on find next/find previous commands (so if your search bar is hidden and you press F3, it will appear). With this change, you no longer need to CTRL + F to remove highlights if it happens that you follow steps described in the issue.

At first I just tried to call `popup_search()` on SEARCH_FIND_NEXT/PREV, but instead of only opening the bar, it also searched for next text, resulting in double search. To remedy this, I created `popup_search_simple()` and `_show_search_simple()`, which do just what their "non-simple" equivalents, minus going to next text. `popup_search_simple()` is called just before `search_next()`.

I really wasn't sure how to do this "correctly", so I hope it's alright >.>

### CODE REVIEW

**Comment by akien-mga** [comment]
I think those "simple" methods add code duplication which is not needed. IMO it's fine (or even expected) that pressing F3 will search for the next entry, even if the search bar is not shown. It's fine by me if we want to change it so that it also shows the search bar, but it should still go to the next entry.

I've tried in the code editor Kate, and the behaviour is as follows:
- pressing Esc hide search bar, pressing it again removes highlight
- pressing F3 searches next entry, shows highlight, doesn't show search bar
- pressing Esc removes highlight

So basically what we have already, with your suggestion from #20643:
>  -make ESC always cancel search highlights even if search field is invisible

**Comment by KoBeWi** [comment]
Every editor does this differently. VS Code uses the behavior I used in PR. Notepad++ doesn't highlight all results, so the problem is non-existent.

The "simple" methods were needed, because with opening the search bar, F3 moved forward past two entries, instead of just one like it should. Only one of methods duplicates code. With `_show_search_simple()`, I'm actually calling it in `_show_search()`, from where I moved two lines that only open the bar. I wanted to do similar thing with second method, but would get a bit hacky, so I decided to repeat few lines.

**Comment by KoBeWi** [comment]
I changed the code to be much simpler, without weird methods.

The branch had however some merge conflicts, so I went through terrible rebase ;_;
Anyways, this PR should be ok now.

**Comment by akien-mga** [comment]
Looks fine to me, though it looks like your rebase somehow squashed your commit with another one from @Kanabenki, which is now being credited as the author. You could fix that with `git commit --reset-author`.

**Comment by KoBeWi** [comment]
I hate rebases ;_;

Fixed.

**Comment by KoBeWi** [comment]
Closing in favor of #31329, as it basically reimplements these changes :/
(it's almost 1000 issues/PRs apart xd)

**Comment by akien-mga** [comment]
>  (it's almost 1000 issues/PRs apart xd)

10000 actually :P
